### DISCUSSION TITLE
has anyone implemented with ionic with react with react-hook-form 6.0.2?

### DISCUSSION THREAD

**Discussion by bradrust** [react-hook-form/react-hook-form#2142] [Q&A] [answered]
I have followed all of the tutorials and blogs however they are all using the older version of react-hook-form. 

Has anyone implemented successfully with 6.0.2?  

I forked a sandbox and change to this new version (because I can't get it to work locally).  

[codesandbox fork of a ionic react example](https://codesandbox.io/s/react-hook-form-ionic-input-components-feopo?file=/src/index.js)

It appears that something is missing from my local implementation, and the sandbox, or perhaps there is a defect.

**Comment by bluebill1049** [comment, marked as answer]
should be a lot easier with `render` prop now. https://react-hook-form.com/api/#Controller

**Reply by bradrust** [reply]
I would have thought so too.   But I can't get it to work... I feel like I added the render correctly but it never seems to ever call my render block.   I updated the codesandbox to at least log during that render.

**Reply by bluebill1049** [reply]
can we start one at a time? focus on the component which having the issue?

**Reply by bradrust** [reply]
I simplified that codesandbox (https://codesandbox.io/s/react-hook-form-ionic-input-components-feopo?file=/src/index.js) so it has only one textfield (IonInput in this case).

You can enter values in that field, clear the field, exit the field, etc... and the render function is never called.   And subsequently, the form is never "valid".  There are no errors displayed.   Just seems like there is no react-hook-form control over the field at all.

I can try whatever you want however I simply don't know enough under the proverbial hood of react, rendering, and all that goodness.

**Reply by bluebill1049** [reply]
I think most of the issue here is related to the props: https://codesandbox.io/s/react-hook-form-ionic-input-components-eor7n?file=/src/index.js you probably want to spend some time with the ironic doc, looks like they have changed few APIs. eg: https://github.com/ionic-team/ionic-framework/issues/17936

**Reply by bluebill1049** [reply]
here are some good examples: https://stackblitz.com/edit/ionic-react-hook-form @bradrust 

https://dev.to/ionic/using-react-hook-form-with-ionic-react-components-update-1463

**Reply by bradrust** [reply]
yep, i saw that he updated that video/example.  tx.
